### PR TITLE
Ensure XML files are opened with UTF-8 encoding

### PR DESCRIPTION
**Pull request by mattwthompson**
### PR Summary:

This PR ensures XML files are opened in UTF-8 mode while being processed. The background here is ... messy and hard to reproduce. What I'm fairly sure of is that my testing infrastructure writes some temporary files (including Foyer XML files) and something might be loaded under an assumption of ASCII encoding. This causes a crash similar to forcing that behavior:

```python
>>> open('oplsaa.xml', encoding='ascii').read()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mattthompson/micromamba/envs/new-models/lib/python3.11/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'ascii' codec can't decode byte 0xce in position 56770: ordinal not in range(128)
```

This is because of the fancy (unicode) $\alpha$ [here](https://github.com/mosdef-hub/foyer/blob/3bb406337493e52d5890dcd6ecee1b1c76ceeb4b/foyer/forcefields/xml/oplsaa.xml#L683):

```python
In [1]: alpha = '\u03b1'

In [2]: alpha
Out[2]: 'α'

In [3]: alpha.encode("utf-8")
Out[3]: b'\xce\xb1'

In [4]: alpha.encode("ascii")
---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
Cell In[4], line 1
----> 1 alpha.encode("ascii")

UnicodeEncodeError: 'ascii' codec can't encode character '\u03b1' in position 0: ordinal not in range(128)
```

AFAICT, Python 3 defaults to `open()`ing files in UTF-8 mode, so this change shouldn't impact anything but esoteric cases like I've run into. (Handwaving more, it's a fixture that crashes _only at collection time_ and only when using `pytest-xdist`.) This would be tricky for me to write a unit test for, but I can take a stab at it if desired.

### PR Checklist
------------
 - [ ] Includes appropriate unit test(s)
 - [ ] Appropriate docstring(s) are added/updated
 - [x] Code is (approximately) PEP8 compliant
 - [ ] Issue(s) raised/addressed?

### CODE REVIEW

File: foyer/forcefield.py [hunk 1]
```diff
@@ -66,7 +66,7 @@
 
     for xml_file in forcefield_files:
         if not hasattr(xml_file, "read"):
-            f = open(xml_file)
+            f = open(xml_file, encoding="utf-8")
```
[the remark this answers is not in the document]
Comment (mattwthompson) [reply foyer/forcefield.py:69]: Heh, this is actually a useful fine

**Comment by chrisjonesBSU** [comment]
I'm trying to rerun the tests, but they keep getting skipped..

If I run the tests locally I'm getting an error:

```
============================================= ERRORS =============================================
__________________________ ERROR collecting foyer/tests/test_trappe.py ___________________________
tests/test_trappe.py:13: in <module>
    TRAPPE_UA = Forcefield(name="trappe-ua")
forcefield.py:522: in __init__
    preprocessed_files = preprocess_forcefield_files(all_files_to_load)
forcefield.py:80: in preprocess_forcefield_files
    xml_contents = f.read()
E   ValueError: I/O operation on closed file.
======================================== warnings summary ========================================
../../../../miniconda3/envs/foyer-dev/lib/python3.12/site-packages/mdtraj/formats/__init__.py:13
  /home/chris/miniconda3/envs/foyer-dev/lib/python3.12/site-packages/mdtraj/formats/__init__.py:13: DeprecationWarning: 'xdrlib' is deprecated and slated for removal in Python 3.13
    from mdtraj.formats.trr import TRRTrajectoryFile

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
---------- generated Nunit xml file: /home/chris/cme/repos/foyer/foyer/test-output.xml -----------
==================================== short test summary info =====================================
ERROR tests/test_trappe.py - ValueError: I/O operation on closed file.
```

It looks like calling `f.close()` under the try/finally statements is causing an issue since a few lines later we call `f.read()` again.

```
  67     for xml_file in forcefield_files:
  68         if not hasattr(xml_file, "read"):
  69             try:
  70                 f = open(xml_file, encoding="utf-8")
  71             finally:
  72                 f.close()
  73
  74             _, suffix = os.path.split(xml_file)
  75         else:
  76             f = xml_file
  77             suffix = ""
  78
  79         # read and preprocess
  80         xml_contents = f.read()
  81         f.close()
```

**Comment by mattwthompson** [comment]
Strange to see .NET stuff float in!

```
------------------------ generated Nunit xml file: /Users/mattthompson/software/foyer/test-output.xml -------------------------

File: environment-dev.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ dependencies:
   - openmm
   - parmed>=3.4.3
   - ele
-  - openff-toolkit >=0.11
+  - openff-toolkit-base >=0.11
```
Comment (chrisjonesBSU) [inline environment-dev.yml:16]: Does this just install a lighter version of openff-toolkit?

**Review by chrisjonesBSU — APPROVED** [review]
LGTM, thanks!

File: environment-dev.yml (hunk 2 above)
Comment (mattwthompson) [reply environment-dev.yml:16]: precisely - and some of the heavier ones/those that are more likely to cause conflicts

https://github.com/conda-forge/openff-toolkit-feedstock/blob/f8d07c4999274f3566062501038537c9ddf1569f/recipe/meta.yaml#L63-L66

Foyer only does some basic interoperability so stuff like partial charge assignment isn't needed. `Molecule.from_smiles` requires RDKit, but that's brought in by mBuild so it worked out here

Comment (chrisjonesBSU) [reply environment-dev.yml:16]: Ohhh ok, that's cool! I didn't know you could create multiple package options like that in a single meta.yml file. I'll have to remember that.

**Review by CalCraven — APPROVED** [review]
Ran tests locally on my end as well here and it looks good, so I'm going to merge this one.

**Comment by mattwthompson** [comment]
Thanks @CalCraven and @chrisjonesBSU!

**Pull request merged**